Makefile

```diff
@@ -15,9 +15,16 @@ REPORT_DIR = report
 SOURCES = $(wildcard *.c)
 OBJECTS = $(SOURCES:%.c=$(BUILD_DIR)/%.o)
 
-all: $(BUILD_DIR)/main
+all: check_dependencies $(BUILD_DIR)/main
 	@echo "$(COLOR_GREEN)============ RUNNING VALGRIND ============$(COLOR_RESET)"
-	valgrind --leak-check=full --show-leak-kinds=all --quiet ./$<
+	valgrind --leak-check=full --show-leak-kinds=all --quiet $(BUILD_DIR)/main
+
+check_dependencies:
+	@echo "$(COLOR_YELLOW)============ CHECKING DEPENDENCIES ============$(COLOR_RESET)"
+	@command -v valgrind > /dev/null || (echo "$(COLOR_RED)Valgrind is not installed. Installing...$(COLOR_RESET)" && sudo apt install -y valgrind)
+	@command -v check > /dev/null || (echo "$(COLOR_RED)Check is not installed. Installing...$(COLOR_RESET)" && sudo apt install -y check)
```

Comment:
**suggestion** (non-blocking): I have removed writing unit tests with `libcheck` library. Instead, I have just used plain old `assert()` functions.

However, I think, we should keep this, if we, in future, write utc using `check` instead of `gtest`.

Reply:
> I think, we should keep this, if we, in future, write utc using check instead of gtest.

Well, yes.